r2d2:
    # Publish all joint states -----------------------------------
    joint_state_controller:
      type: joint_state_controller/JointStateController
      publish_rate: 50

    # Velocity Controllers ---------------------------------------
    roll_joint_velocity_controller:
      type: "velocity_controllers/JointVelocityController"
      joint: "roll_joint"
      pid:
        p: 1.0
        i: 1.0
        d: 0.0
    wheel_arm1_joint_velocity_controller:
      type: "velocity_controllers/JointVelocityController"
      joint: "wheel_arm1_joint"
      pid:
        p: 1.0
        i: 1.0
        d: 0.0
    wheel_arm2_joint_velocity_controller:
      type: "velocity_controllers/JointVelocityController"
      joint: "wheel_arm2_joint"
      pid:
        p: 1.0
        i: 1.0
        d: 0.0
    gazebo_ros_control:
      pid_gains:
        roll_joint:
          p: 1.0
          i: 1.0
          d: 0.0
        wheel_arm1_joint:
          p: 1.0
          i: 1.0
          d: 0.0
        wheel_arm2_joint:
          p: 1.0
          i: 1.0
          d: 0.0
        incline_joint:
          p: 100.0
          i: 10.0
          d: 1.0

    # To add more just add them here as the first one

    # Position Controllers ---------------------------------------
    incline_joint_position_controller:
      type: "position_controllers/JointPositionController"
      joint: "incline_joint"
      pid: {p: 1200.0, i: 120.0, d: 4.0}
    base_cylinder_joint_position_controller:
      type: effort_controllers/JointPositionController
      joint: base_cylinder_joint
      pid: {p: 1300.0, i: 250.0, d: 0.0}
    base_head_joint_position_controller:
      type: effort_controllers/JointPositionController
      joint: base_head_joint
      pid: {p: 1.0, i: 1.0, d: 0.0}
    shoulder1_joint_position_controller:
      type: effort_controllers/JointPositionController
      joint: shoulder1_joint
      pid: {p: 100.0, i: 10.0, d: 1.0}
    coude1_joint_position_controller:
      type: effort_controllers/JointPositionController
      joint: coude1_joint
      pid: {p: 800.0, i: 80.0, d: 3.0}
    shoulder2_joint_position_controller:
      type: effort_controllers/JointPositionController
      joint: shoulder2_joint
      pid: {p: 100.0, i: 10.0, d: 1.0}
    coude2_joint_position_controller:
      type: effort_controllers/JointPositionController
      joint: coude2_joint
      pid: {p: 800.0, i: 80.0, d: 3.0}

controller_manager:
  update_rate: 50